### PR TITLE
feat(edit-ema): Sync the URL from Angular to the iframe

### LINKED ISSUE #26713: Edit EMA: Syncing URL and Query Params for Languages from Angular to the Iframe
### User Story
As a dotCMS user, I want to be able to sync the URL and query params for languages from Angular to the iframe, so I can easily switch between languages.

### Acceptance Criteria
- Implement the angular routing system
- URL and query params for languages should be in sync

### Proposed Objective
Core Features

### Proposed Priority
Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.:
N/A

### Assumptions & Initiation Needs
- Angular routing should be properly implemented

### Quality Assurance Notes & Workarounds
N/A

### Sub-Tasks & Estimates:
- Implement the angular routing system (2 hrs)
- Sync URL and query params for languages (3 hrs)

### PR DESCRIPTION
**Pull request by fmontes**
### Proposed Changes
* Add a toolbar with two selects: lang and pages and this changes the Angular url
* Use angular router to manipulate the url
* Pass the localhost synced url to the iframe

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

File: core-web/libs/portlets/dot-ema/src/lib/components/edit-ema-toolbar/edit-ema-toolbar.component.html
Comment (fmontes) [inline core-web/libs/portlets/dot-ema/src/lib/components/edit-ema-toolbar/edit-ema-toolbar.component.html:1]: This is just a placeholder.

File: dotCMS/build.gradle
Comment (zJaaal) [inline dotCMS/build.gradle:1]: Remove this

**Review by zJaaal — CHANGES_REQUESTED** [review]

**Review by zJaaal — APPROVED** [review]

File: core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.ts [hunk 1]
```diff
@@ -1,11 +1,88 @@
...
+        },
+        {
+            name: 'Page Two',
+            value: 'page-two'
+        }
+    ];
+
+    route = inject(ActivatedRoute);
+    router = inject(Router);
+    store = inject(EditEmaStore);
```
Comment (rjvelazco) [inline core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.ts:45]: Should these porperties be `private`?

File: core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.ts [hunk 2]
```diff
@@ -1,11 +1,88 @@
...
+
+    iframeUrl$ = this.store.iframeUrl$;
+    language_id$ = this.store.language_id$;
+    title$ = this.store.pageTitle$;
+    url$ = this.store.url$;
```
Comment (rjvelazco) [inline core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.ts:50]: Do we need separate observables? Or can we use just one?

File: ema/nextjs/src/app/[[...slug]]/page.js [hunk 3]
```diff
@@ -21,8 +22,12 @@ async function getPage(url) {
     return res.json();
 }
 
-export default async function Home({ params }) {
-    const data = await getPage(params.slug);
+export default async function Home({ searchParams, params }) {
+    console.log(params.slug);
```
Comment (rjvelazco) [inline ema/nextjs/src/app/[[...slug]]/page.js:26]: Do we need this?

File: dotCMS/build.gradle [hunk 4]
```diff
@@ -265,7 +265,7 @@ dependencies {
 
     starter group: 'com.dotcms', name: 'starter', version: 'empty_20230710', ext: 'zip'
     //Uncomment this line if you want to download the starter that comes with data
-    //starter group: 'com.dotcms', name: 'starter', version: '20230712', ext: 'zip'
+    starter group: 'com.dotcms', name: 'starter', version: '20230712', ext: 'zip'
```
Comment (rjvelazco) [inline dotCMS/build.gradle:268]: Comment this

**Review by rjvelazco — CHANGES_REQUESTED** [review]

File: core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.ts [hunk 5]
```diff
@@ -1,11 +1,88 @@
...
+            case 'language_id':
+                this.store.setLanguage(value);
+                break;
+
+            case 'url':
+                this.store.setUrl(value);
+                break;
+        }
+
+        this.router.navigate([], {
```
Comment (KevinDavilaDotCMS) [inline core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.ts:81]: Nice!!

File: core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.ts (hunk 1 above)
Comment (fmontes) [reply core-web/libs/portlets/dot-ema/src/lib/feature/dot-ema.component.ts:45]: yes they should!

**Review by rjvelazco — APPROVED** [review]

**Review by KevinDavilaDotCMS — APPROVED** [review]

**Pull request merged**